### PR TITLE
[FIXED] MaxPending > MaxInt32 causes client to be disconnected

### LINKED ISSUE #1118: Options.MaxPending is defined as int64 but used as int32
- [x] Defect
 - [ ] Feature Request or Change Proposal
 
## Defects 

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [ ] Included `gnatsd -DV` output
 - [x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)
 
#### Steps or code to reproduce the issue: 
```
package main

import (
	"math"
	"time"

	"github.com/nats-io/nats-server/v2/server"
	natsc "github.com/nats-io/nats.go"
)

func main() {
	svc, _ := server.NewServer(&server.Options{
		MaxPending: math.MaxInt32 + 1,
	})
	svc.ConfigureLogger()
	go func() {
		time.Sleep(1 * time.Second)
		opts := natsc.GetDefaultOptions()
		conn, _ := opts.Connect()

		_, _ = conn.Subscribe("*", nil)
		conn.Flush()
	}()
	svc.Start()
}
```
 
#### Expected result:
It's less than math.MaxInt64 so it should run without any issues.
 
#### Actual result:
 Exits with this error: 
`[75524] [INF] 127.0.0.1:52013 - cid:1 - Slow Consumer Detected: MaxPending of -503316480 Exceeded`
 
#### Proposed Change:
 Either change `Options.MaxPending` to `int32` or use it as `int64` in the codebase.

#### Who Benefits From The Change(s)?
 Everyone

### PR DESCRIPTION
**Pull request by kozlovic**
* Changed some of client.outbound fields to int64.
* Moved fields around to minimize size of struct (checked with unsafe.Sizeof())
* Checked benchmark results before/after
* Added test

Resolves #1118

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/client.go [hunk 1]
```diff
@@ -217,17 +217,17 @@ type outbound struct {
 	p   []byte        // Primary write buffer
 	s   []byte        // Secondary for use post flush
 	nb  net.Buffers   // net.Buffers for writev IO
-	sz  int32         // limit size per []byte, uses variable BufSize constants, start, min, max.
-	sws int32         // Number of short writes, used for dynamic resizing.
-	pb  int32         // Total pending/queued bytes.
-	pm  int32         // Total pending/queued messages.
+	sz  int64         // limit size per []byte, uses variable BufSize constants, start, min, max.
```
Comment (derekcollison) [inline server/client.go]: I think this one can still be int32.

File: server/client.go [hunk 2]
```diff
@@ -217,17 +217,17 @@ type outbound struct {
...
+	pb  int64         // Total pending/queued bytes.
+	mp  int64         // Snapshot of max pending for client.
+	lwb int64         // Last byte size of Write.
```
Comment (derekcollison) [inline server/client.go]: This one as well.

Comment (kozlovic) [reply server/client.go]: If you can allow a write to be more than int32, then how this can stay a int32?

Comment (kozlovic) [reply server/client.go]: because we limit a write to int32?

Comment (derekcollison) [reply server/client.go]: the sz will be int32, it has a max which we do not exceed. I doubt lwb would ever be over it but we could put in a check to make sure.

**Comment by derekcollison** [comment]
I would prefer just mp be changed to int64.

**Comment by kozlovic** [comment]
Not even pb (total pending/queued bytes)???

**Comment by derekcollison** [comment]
Apologies, pb would have to match mp.

**Comment by kozlovic** [comment]
There will be a lot of casting going on...

**Comment by derekcollison** [comment]
ok we can revisit later when we put connections on a diet. Just hate to keep adding to them unnecessarily.

File: server/client.go (hunk 1 above)
Comment (kozlovic) [reply server/client.go]: I have reverted this change.

File: server/client.go (hunk 2 above)
Comment (kozlovic) [reply server/client.go]: Ok, changed just pb and mp to int64 (but moved some fields to have more compact structure).

**Comment by kozlovic** [comment]
@derekcollison Changes back some of the int64 to int32 and added casting where needed. Tried to keep the structure as compact as possible by moving some of the fields around.

**Pull request merged**